### PR TITLE
CI: blocking lint and more lint targets

### PR DESCRIPTION
**Pull request by jbertran**
#### WHAT

Make lint blocking and fix python linting issues.

Add linting targets:
* YAML
* Shell

#### WHY

Blocking lint tests is good. So is more linters.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -16,6 +16,15 @@ deps =
 commands =
     bash -c "bashate **/*.sh"
 
+[testenv:lint-yaml]
+whitelist_externals = bash
+basepython = python3.6
```
Comment (slaperche-scality) [inline tox.ini]: Do you need to specify `basepython`? It's not inherited from `testenv`?

**Review by slaperche-scality — DISMISSED** [review]

File: salt/_states/containerd.py [hunk 2]
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
             ret['result'] = True
         else:
             ret['comment'] = 'Failed to pull image'
+        return ret
+
+    return ret
```
Comment (charlesprost) [inline salt/_states/containerd.py:118]: Maybe `return ret` only once?

File: salt/_states/containerd.py [hunk 3]
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
...
+
 
+def manage_archive_path(name, archive_path, ret):
+    '''Pull image from archive at the specified path
```
Comment (charlesprost) [inline salt/_states/containerd.py]: Missing endpoint?

File: salt/_states/containerd.py [hunk 4]
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
...
+
+    name
+        Tag or digest of the image
+    archive_path
+        path to the local Docker archive of the image to load if
```
Comment (charlesprost) [inline salt/_states/containerd.py]: capitalization of `path`?

File: salt/_states/containerd.py [hunk 5]
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
...
+                'old': {},
+                'new': result,
+            },
+        })
+        ret['comment'] = 'Imported archive'
+        ret['result'] = True
+    else:
+        ret['comment'] = 'Failed to import archive'
+
+    return ret
```
Comment (charlesprost) [inline salt/_states/containerd.py]: Same for `return ret`?

**Review by charlesprost — DISMISSED** [review]
minor issues to be fixed

File: make/lint.mk [hunk 6]
```diff
@@ -5,5 +5,7 @@
 
 lint:
 	tox -e lint-salt
+	tox -e lint-yaml
+	tox -e lint-shell
 
 .PHONY: lint-local
```
Comment (NicolasT) [inline make/lint.mk]: There's no `lint-local` target?

File: salt/_modules/containerd.py [hunk 7]
```diff
@@ -7,7 +7,7 @@
 
 import salt.utils.path
 
-log = logging.getLogger(__name__)
+LOG = logging.getLogger(__name__)
```
Comment (NicolasT) [inline salt/_modules/containerd.py]: Most other (upstream) Salt modules use `log` i.s.o. `LOG`. Maybe this should be an exception in the linter.

File: salt/_states/containerd.py [hunk 8]
```diff
@@ -114,5 +90,45 @@ def image_managed(name, archive_path=None):
             ret['result'] = True
         else:
             ret['comment'] = 'Failed to pull image'
+        return ret
+
+    return ret
+
 
+def manage_archive_path(name, archive_path, ret):
```
Comment (NicolasT) [inline salt/_states/containerd.py]: This naming seems not in line with other states. `manage` is active (i.e. a module).

If the intent is for this to be not 'public API of the state module' make it `_manage_archive_path` or something, though even then, even utility procedures in a state should be... a state-like thing.

Comment (NicolasT) [inline salt/_states/containerd.py]: What's this `ret` thing?

File: tox.ini [hunk 9]
```diff
@@ -7,14 +7,30 @@ skip_install = true
 setenv =
     VIRTUALENV_NO_DOWNLOAD=1
 
+[testenv:lint-shell]
+whitelist_externals = bash
+basepython = python3.6
+skip_install = true
+deps =
+    -r{toxinidir}/lint-requirements.txt
```
Comment (NicolasT) [inline tox.ini]: Now you're installing all lint deps in 3 different venvs. Either

- List linter-specific deps here (I think that'd be my preference)
- Set up Tox to re-use the same env between the 3 jobs

File: salt/_states/containerd.py (hunk 2 above)
Comment (jbertran) [reply salt/_states/containerd.py:118]: I really dislike the control flow of these dict-filling steps. Yes, you're absolutely right.

File: salt/_modules/containerd.py (hunk 7 above)
Comment (slaperche-scality) [reply salt/_modules/containerd.py]: On the other hand, it's pretty common to have UPPERCASE for global variable.

So depends if we want clean code or Salt-like code here.

But if we add an exception to the linter, it should be only for Salt code, as we probably don't want these kind of weird decision to be applied to Python script in general

File: salt/_states/containerd.py (hunk 3 above)
Comment (jbertran) [reply salt/_states/containerd.py]: Done

File: salt/_states/containerd.py (hunk 4 above)
Comment (jbertran) [reply salt/_states/containerd.py]: Done

File: salt/_modules/containerd.py (hunk 7 above)
Comment (jbertran) [reply salt/_modules/containerd.py]: Leaving it as `log` for now since I'm dropping Python linting

File: make/lint.mk (hunk 6 above)
Comment (jbertran) [reply make/lint.mk]: Indeed :o

File: salt/_states/containerd.py (hunk 8 above)
Comment (jbertran) [reply salt/_states/containerd.py]: Reverting this since I'm dropping Python linting.

Comment (jbertran) [reply salt/_states/containerd.py]: Reverted. Getting lost in salt return dict side effects.

File: tox.ini (hunk 9 above)
Comment (jbertran) [reply tox.ini]: lint deps are minimal, so linter-specific it is.

File: tox.ini (hunk 1 above)
Comment (jbertran) [reply tox.ini]: Fixed

File: salt/_modules/containerd.py (hunk 7 above)
Comment (slaperche-scality) [reply salt/_modules/containerd.py]: Why reverting that change? You'll have to reintroduce it when we re-enable linting anyway, so you could keep it.

Comment (jbertran) [reply salt/_modules/containerd.py]: It's trivial enough to reproduce, and if I want to see if I can work in the Salt-specific exception.

**Review by slaperche-scality — DISMISSED** [review]

File: tox.ini [hunk 10]
```diff
@@ -2,21 +2,20 @@
...
 
-[testenv:lint-salt]
-whitelist_externals = bash
-basepython = python3.6
-skip_install = true
+[testenv:lint-shell]
+deps =
+    bashate==0.6.0
+commands =
+    bash -c "bashate **/*.sh"
```
Comment (gdemonet) [inline tox.ini:15]: Would be nice to add `**/*.sh.in` as well, so we can lint e.g. `scripts/bootstrap.sh.in`

**Review by gdemonet — DISMISSED** [review]

File: tox.ini (hunk 10 above)
Comment (jbertran) [reply tox.ini:15]: Done. We ignore long line warnings for `.sh.in`, since they're needed by some of the heredocs included.

**Comment by jbertran** [comment]
@bert-e approve

**Review by gdemonet — APPROVED** [review]

**Pull request merged**